packages/instantsearch.js/src/lib/utils/__tests__/hydrateSearchClient-test.ts

```diff
@@ -147,4 +147,24 @@ describe('hydrateSearchClient', () => {
       } as unknown as InitialResults);
     }).not.toThrow();
   });
+
+  it('should not throw if state or results are missing', () => {
```

Comment:
in think in Vue we might have been relying on this (maybe a separate error written there) to ensure you're actually passing _something_ to the function